### PR TITLE
#26 Refactor Sizing!

### LINKED ISSUE #26: Refactoring - Refactor fixed widths and heights to use grids
### Description

When I first made this I really didnt know much about tkinter and how the grid system worked, which led to alot of setting fixed widths and heights for everything, which I was able to get away with because we use a fixed width and height screen. But if the engineers decide they want to shrink or make the screen bigger, it would require a ton of change, but this can be avoided by using grids and sticky values appropriately, as well as making things relative to height and width variables, instead of magic numbers.

### Acceptance Criteria

Remove as many fixed widths and heights as possible

### Proposed Solution

Ideally we would have a universal height and width variable, that represents the height and width of whatever screen wants to display it, and then everything would scale based off of that variable. I.e if you want a frame to take up a third of the screen, you would set it to width/3 or height/3 depending on what ur looking for. 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Removed Packs entirely from the codebase (dont know why I thought we needed it) 
Now dynamically resize regardless of screen size(can make it full screen on ur comp if u want) 
Made a single width and height variable that all the code references
uhh probably other stuff i cant remember

## Screenshots
Screen Shot 2023-03-18 at 8 28 10 PM
Screen Shot 2023-03-18 at 8 27 59 PM

Screen Shot 2023-03-18 at 7 46 11 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #26  (issue #)

### CODE REVIEW

File: models/mock_model.py [hunk 1]
```diff
@@ -87,16 +88,16 @@ def on_press(self, key):
                 self.enter = 1
             case Key.right:
                 self.forward = 1
-                self.mode_index = (self.mode_index + 1) if self.mode_index < 6 else 0
+                self.mode_index = (self.mode_index + 1) if self.mode_index < 5 else 0
             case Key.left:
                 self.backward = 1
-                self.mode_index = (self.mode_index - 1) if self.mode_index > 0 else 6
+                self.mode_index = (self.mode_index - 1) if self.mode_index > 0 else 5
```
Comment (anthonybernardi) [inline models/mock_model.py]: this code is bad, just do the length of the modes or something. this is not good to change every time we add or remove a mode (and pls fix everywhere)

File: nero_view.py [hunk 2]
```diff
@@ -40,66 +44,74 @@ def __init__(self) -> None:
...
 
     def update_mode(self):
-        if self.model.get_BMS_state() is not None and self.model.get_BMS_state() >= 2:
-            self.current_mode = self.modes[6]
+        if self.is_debug:
+            self.current_screen = self.debug_screen
         else:
-            self.current_mode = self.modes[self.mode_index]
-        self.current_mode.tkraise()
+            self.current_screen = self.modes[self.mode_index]
```
Comment (anthonybernardi) [inline nero_view.py:79]: a screen can be a mode too? this doesn't make any sense

File: nero_view.py [hunk 3]
```diff
@@ -40,66 +44,74 @@ def __init__(self) -> None:
...
         # create the modes that the container will hold
-        self.modes = []
+        self.modes: List[Mode] = []
         self.mode_index = 0
-
-        self.header = Header(parent=container)
-        self.header.place(x=0, y=0, relwidth=1, relheight=0.05)
-        for mode_class in (OffMode, PitLaneMode, EfficiencyMode, SpeedMode, DebugMode, ReverseMode, ChargingMode):
-            # for mode_class in (EfficiencyMode, DebugMode, ChargingMode):
+        for mode_class in (OffMode, PitLaneMode, EfficiencyMode, SpeedMode, ReverseMode, ChargingMode):
```
Comment (anthonybernardi) [inline nero_view.py]: we should maybe make this list a constant at the top of the file

File: nero_view.py [hunk 4]
```diff
@@ -40,66 +44,74 @@ def __init__(self) -> None:
...
+        self.update_mode()
+        self.after(1, self.update_mode_index)
+
     def check_can(self):
         self.model.check_can()
         self.after(1, self.check_can)
 
     def update_current_page(self):
-        self.current_mode.current_page.update()
+        self.current_screen.current_page.update()
```
Comment (anthonybernardi) [inline nero_view.py:90]: we dont need current_screen at all if debug """screen""" is just a mode anyways

you're overconforming to their definitions (and their definitions are pretty bad and silly)

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
ui stuff is good

File: nero_view.py (hunk 2 above)
Comment (Peyton-McKee) [reply nero_view.py:79]: we said in slack that the current screen can either be debug or the current mode, and debug overrides the current mode to be displayed as the current screen

File: nero_view.py (hunk 4 above)
Comment (Peyton-McKee) [reply nero_view.py:90]: we need two variables to track the current mode and the current page being displayed, so i called it the current screen, we need both of them to exist at the same time

File: nero_view.py [hunk 5]
```diff
@@ -1,12 +1,9 @@
...
+from typing import List
 from modes.debug_mode.debug_mode import DebugMode
-from modes.efficiency_mode.efficiency_mode import EfficiencyMode
-from modes.off_mode.off_mode import OffMode
-from modes.charging_mode.charging_mode import ChargingMode
-from modes.pit_lane_mode.pit_lane_mode import PitLaneMode
-from modes.reverse_mode.reverse_mode import ReverseMode
-from modes.speed_mode.speed_mode import SpeedMode
+from modes.mode import Mode
+from models.mock_model import MODES
```
Comment (anthonybernardi) [inline nero_view.py]: the mock model should not be the source of truth for the list of modes

that should probably be here or in some file dedicated to constants if we want

File: nero_view.py [hunk 6]
```diff
@@ -40,66 +39,69 @@ def __init__(self) -> None:
...
+            self.current_screen = self.debug_screen
         else:
-            self.current_mode = self.modes[self.mode_index]
-        self.current_mode.tkraise()
+            self.current_screen = self.modes[self.mode_index]
+        self.current_mode = self.modes[self.mode_index]
+        self.current_screen.tkraise()
         self.header.tkraise()
 
+    def update_mode_index(self):
```
Comment (anthonybernardi) [inline nero_view.py]: update_mode_index and update_mode should just be one function called update_mode (this file only has one real line of code so just put it in the other one)

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
im good with this after addressing these two small comments

**Review by anthonybernardi — APPROVED** [review]
swag

**Pull request merged**